code/game/machinery/metaldetector.dm

```diff
@@ -0,0 +1,87 @@
+GLOBAL_LIST_INIT(metal_detector_items, typecacheof(list(
+	/obj/item/weapon/gun,
+	/obj/item/weapon/material,
+	/obj/item/weapon/melee,
+	/obj/item/device/transfer_valve,
+	/obj/item/weapon/grenade/,
+	/obj/item/ammo_casing/,
+	/obj/item/ammo_magazine
+	)))
+
+/obj/machinery/metal_detector
+	name = "metal detector"
+	desc = "An advanced metal detector used to detect weapons."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "metal_detector"
+	anchored = TRUE
+	req_one_access = list(access_security, access_heads)
+	idle_power_usage = 100
+
+	plane = MOB_PLANE
+	layer = ABOVE_MOB_LAYER
+
+	var/next_announcement_time	// so this doesn't get spammy.
+
+	var/alarm_delay = 2000
+
+/obj/machinery/metal_detector/New()
+	..()
+	next_announcement_time = get_game_time() - alarm_delay
+
+	component_parts = list()
+	component_parts += new /obj/item/weapon/circuitboard/metal_detector(src)
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/weapon/stock_parts/console_screen(src)
+	RefreshParts()
+
+/obj/machinery/metal_detector/examine(mob/user)
+	if((stat & NOPOWER))
+		to_chat(user, "An advanced metal detector used to detect weapons. It is currently unpowered.")
+	else if((stat & BROKEN))
+		to_chat(user, "An advanced metal detector used to detect weapons. It seems to be broken.")
+	else
+		to_chat(user, "An advanced metal detector used to detect weapons.")
+
+/obj/machinery/metal_detector/attackby(obj/item/W, mob/usr)
+
+	if(default_deconstruction_screwdriver(usr, W))
+		return
+
+	if(default_deconstruction_crowbar(usr, W))
+		return
+
+	..()
+
+/obj/machinery/metal_detector/Crossed(var/mob/living/M)
+	..()
+
+	if((stat & (NOPOWER|BROKEN)))
+		return
+
+	if(!istype(M))
+		return
+	if(allowed(M))
+		return
+
+	var/list/items_to_check = M.GetAllContents()
```

Comment:
Again with the lack of recursion limits, I'm a little concerned that someone with a large enough inventory could cause actual lag spikes by just walking through a metal detector

Reply:
Seconded.

Reply:
Would the recursion limits be the thing I had prior removed out of stupidity and now reinstated, or is there something that must be changed here?

Reply:
With `helpers/unstorted.dm` untouched, this will use the default depth of 5.
The reason that a lack of limitation is concerning is because, if someone has boxes stuffed inside backpacks stuffed inside bags of holding ad nauseam, etc., then getAll Contents will iterate over every single item, which for inventories with a large number of items, could potentially cause the server to hang.